### PR TITLE
fix mbo button in billing manager

### PR DESCRIPTION
**Pull request by stephenwaite**
#### Short description of what this resolves:
fix for mbo button on billing page

#### Changes proposed in this pull request:
use `POST` instead of `SESSION` and set the `encounter` and `pid`

### CODE REVIEW

**Comment by stephenwaite** [comment]
while debugging noticed that the `$_SESSION` values were not passed from the new script to the save script so had to switch to `$_POST`

not sure if there was a session close when the new script finished which prevented them from being passed

File: interface/forms/misc_billing_options/new.php [hunk 1]
```diff
@@ -13,27 +13,35 @@
...
+require_once("$srcdir/encounter.inc");
 
 use OpenEMR\Common\Csrf\CsrfUtils;
 use OpenEMR\Common\Session\SessionUtil;
 use OpenEMR\Core\Header;
 use OpenEMR\OeUI\OemrUI;
 
 if (isset($_REQUEST['isBilling'])) {
+    $is_from_billing_manager = true;
     $pid = $_SESSION['billpid'] = $_REQUEST['pid'];
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php:37]: This should sort out the session issue (can not directly set a session variable; need to use this fancy function :) )
```
$pid = $_REQUEST['pid'];
SessionUtil::setSession('billpid', $pid);
```

File: interface/forms/misc_billing_options/new.php [hunk 2]
```diff
@@ -13,27 +13,35 @@
...
+    if ($pid != $_SESSION["pid"]) {
+        setpid($pid);
+    }
     $encounter = $_SESSION['billencounter'] = $_REQUEST['enc'];
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php:41]: This should sort out the session issue (can not directly set a session variable; need to use this fancy function :) )
(replace above line with below)
```
$encounter = $_REQUEST['enc'];
SessionUtil::setSession('billencounter', $encounter);
```

File: interface/forms/misc_billing_options/new.php [hunk 3]
```diff
@@ -13,27 +13,35 @@
...
 use OpenEMR\Common\Session\SessionUtil;
 use OpenEMR\Core\Header;
 use OpenEMR\OeUI\OemrUI;
 
 if (isset($_REQUEST['isBilling'])) {
+    $is_from_billing_manager = true;
     $pid = $_SESSION['billpid'] = $_REQUEST['pid'];
+    if ($pid != $_SESSION["pid"]) {
+        setpid($pid);
+    }
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php:40]: still need this block

File: interface/forms/misc_billing_options/new.php [hunk 4]
```diff
@@ -13,27 +13,35 @@
...
     $encounter = $_SESSION['billencounter'] = $_REQUEST['enc'];
+    if ($encounter != $_SESSION["encounter"]) {
+        setencounter($encounter);
+    }
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php:44]: still need this block

File: interface/forms/misc_billing_options/new.php [hunk 5]
```diff
@@ -13,27 +13,35 @@
...
+require_once("$srcdir/pid.inc");
+require_once("$srcdir/encounter.inc");
 
 use OpenEMR\Common\Csrf\CsrfUtils;
 use OpenEMR\Common\Session\SessionUtil;
 use OpenEMR\Core\Header;
 use OpenEMR\OeUI\OemrUI;
 
 if (isset($_REQUEST['isBilling'])) {
+    $is_from_billing_manager = true;
```
Comment (bradymiller) [inline interface/forms/misc_billing_options/new.php:36]: should hopefully no longer need above line or other code changes below

File: interface/forms/misc_billing_options/new.php (hunk 2 above)
Comment (stephenwaite) [reply interface/forms/misc_billing_options/new.php:41]: oh jeez, thank you :)

**Pull request merged**